Libraries/JavaScriptAppEngine/Initialization/exceptionOccurHint.js

```diff
@@ -0,0 +1,34 @@
+/**
+ * Copyright (c) 2015-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @providesModule exceptionOccurHint
+ */
+
+'use strict';
+
+const SourceMapsCache = require('SourceMapsCache');
+const parseErrorStack = require('parseErrorStack');
+
+function exceptionOccurHint() {
+  const stack =  parseErrorStack(new Error());
+  const renderFrame = stack.find((frame) => frame.methodName.endsWith('render'));
+
+  if (renderFrame) {
+    return SourceMapsCache.getSourceMaps().then(sourceMaps => {
+      const prettyStack = parseErrorStack({stack: [renderFrame]}, sourceMaps);
+      const prettyFrame = prettyStack[0];
+      const fileParts = prettyFrame.file.split('/');
+      const fileName = fileParts[fileParts.length - 1];
+      return `At: ${fileName }:${prettyFrame.lineNumber}`;
+    });
+  } else {
+    return new Promise((resolve, reject) => resolve(null));
```

Comment:
So if renderFrame doesn't exist, let's just return the file/line where the console.warn is, and also include the pretty stack after the whole error message.